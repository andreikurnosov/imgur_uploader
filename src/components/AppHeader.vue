<template>
  <div class="ui secondary pointing menu">
    <router-link to="/" class="active item">Image Storage</router-link>
    <div class="right menu">
      <a href="#" class="ui item" @click="login">Login üçï</a>
    </div>
  </div>
</template>

<script>
import { mapActions } from 'vuex'

export default {
  name: 'AppHeader',
  methods: {
    ...mapActions(['login'])
  }
}
</script>

<style lang="scss" scoped></style>
