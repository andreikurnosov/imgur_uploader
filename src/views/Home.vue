<template>
  <div class="app-container">
    <AppHeader />
    <div class="ui container">
      <router-view />
    </div>
  </div>
</template>

<script>
import AppHeader from '@/components/AppHeader.vue'
export default {
  name: 'Home',
  components: {
    AppHeader
  }
}
</script>
<style scoped>
.app-container {
  width: 70vw;
  height: 100vh;
  margin: auto;
  background-color: aliceblue;
  box-shadow: 0px 0px 5px grey;
}
</style>
