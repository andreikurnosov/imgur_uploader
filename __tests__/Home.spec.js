import { shallowMount, createLocalVue } from '@vue/test-utils'
import Home from '@/views/Home'
import VueRouter from 'vue-router'
import AppHeader from '@/components/AppHeader'

const localVue = createLocalVue()
localVue.use(VueRouter)

const router = new VueRouter()

describe('Home', () => {
  const build = () => {
    const wrapper = shallowMount(Home, {
      localVue,
      router
    })

    return {
      wrapper,
      appHeader: () => wrapper.findComponent(AppHeader)
    }
  }

  it('renders the component', () => {
    const { wrapper } = build()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders main child components', () => {
    const { appHeader } = build()

    expect(appHeader().exists()).toBe(true)
  })
})
