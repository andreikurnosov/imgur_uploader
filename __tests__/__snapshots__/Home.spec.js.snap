// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders the component 1`] = `
<div class="app-container">
  <appheader-stub></appheader-stub>
  <div class="ui container">
    <router-view-stub name="default"></router-view-stub>
  </div>
</div>
`;
